# Análisis de sueldos con Python

## Data

* [2021.2 - sysarmy - Encuesta de remuneración salarial Argentina
](https://docs.google.com/spreadsheets/d/1-ZRznwS4TK74o90aOiCKS5SiXxUQ2buN1wxZIMHQmzQ/edit#gid=557755710) 
* [2021.2 - sysarmy - Encuesta de remuneración salarial Latam
](https://docs.google.com/spreadsheets/d/1BkBNt1MHVS7DeIlpgmK9l6krtSQ5t_olRhlcyxMeKy0/edit#gid=557755710)

1. Descargar, leer y unir en una sola estructura de datos. ¿Qué estructura elegiste y por qué?

In [1]:
path_to_file_1 = "2021.2_-_sysarmy_-_Encuesta_de_remuneracion_salarial_Argentina_-_Argentina.csv"
path_to_file_2 = "2021.2_-_sysarmy_-_Encuesta_de_remuneracion_salarial_Latam_-_Latam.csv"

import csv

raw_cases_ARG = []
raw_cases_LATAM = []
cases_list = []


with open(f"{path_to_file_1}", "r") as f1:
    reader = csv.reader(f1)
    for row in reader:
        raw_cases_ARG.append(row)


with open(f"{path_to_file_2}", "r") as f2:
    reader = csv.reader(f2)
    for row in reader:
        raw_cases_LATAM.append(row)
        

del raw_cases_ARG[:10]
del raw_cases_LATAM[:11]



for case in raw_cases_ARG:
    cases_list.append({
    "Region": "",
    "Tipo contrato": "",
    "Salario Bruto": 0 ,
    "Salario Neto" : 0 ,
    "Pagos en dólares": 0 ,
    "Valor de dolar tomado": 0 ,
    "Conformidad con sueldo": 0 ,
    "Conformidad con sueldo último semestre": 0 ,
    "Bono": "",
    "A qué está atado el bono": "",
    "Ajustes por inflación 2021": "",
    "% del ajuste": 0,
    "Mes de Ajuse": 0,
    "Rol": "",
    "Años de experiencia": 0,
    "Años en la empresa": 0,
    "Años en el puesto": 0,
    "Gente a cargo": 0,
    "Plataformas": 0,
    "Lenguajes de programación - tecnologías": "",
    'Frameworks - herramientas - librerías': "",
    "Bases de datos": '',
    "QA / Testing": "",
    "IDEs": "",
    "Cantidad de personas en org": 0,
    "Org con oficina en mi ciudad": "",
    "Actividad principal": "",
    "eNPS": 0,
    "Calificación de políticas de diversidad e inclusión": 0,
    "Beneficios extra": "",
    "Salir o seguir Estudios":"",
    "Nivel de estudios alcanzado": "",
    "Estado": "",
    "Carrera": "",
    "Universidad": "",
    "Cursos de especialización": "",
    "¿Contribuís a proyectos open source?": "",
    "¿Programás como hobbie?": "",
    "Salir o seguir Guardias": "",
    "¿Tenés guardias?": "",
    "Cuánto cobrás por guardia": "",
    "¿Porcentaje, bruto o neto?": "",
    "Salir o seguir Seguridad Laboral": "",
    "¿Situaciones de violencia laboral?": "",
    "¿Tenés algún tipo de discapacidad?": "",
    "Dificultad para conseguir trabajo por discapacidad": "",
    "Salir o seguir Pandemia": "",
    "¿Cómo venís llevando la pandemia?": "",
    "¿Tenés hijos/as menores de edad?": "",
    "¿Con cuántas personas estás conviviendo?": "",
    "¿Con quiénes convivís?": "",
    "¿Tenés un espacio dedicado para el trabajo?": "",
    "¿Tenés que compartir tu equipo de trabajo con alguien?": "",
    "¿Qué tipo de cuarentena hiciste / estás haciendo?": "",
    "¿Cambió tu situación laboral a raíz de la pandemia?": "",
    "Apoyo de empresa/organización en esta situación?": "",
    "¿Cómo se vio afectada tu empresa/organización?": "",
    "¿Instauraron algún beneficio nuevo?": "",
    "Edad": "",
    "Identidad Sex": "",        
        
})


for i, case in enumerate(raw_cases_ARG):
    try:
        for j, key  in enumerate(cases_list[i].keys()):
            cases_list[i][key] = raw_cases_ARG[i][j]
    except IndexError:
        print("index error en línea "+ str(raw_cases_ARG.index(raw_cases_ARG[i])) + str(raw_cases_ARG[i]))
    

for case in cases_list:
    case.update({"Pais": "Argentina"})
    
aux_cases_list = []
    
for case in raw_cases_LATAM:
    aux_cases_list.append({"Pais": case[0]})
    
countries_set = set()

for case in raw_cases_LATAM:
        countries_set.add(case[0])


countries_list = list(countries_set)
countries_list.sort()
cites_cases_list = []


for case in raw_cases_LATAM:
    for country in countries_list:
        if case[0] == country:
            cites_cases_list.append ({"Region": case[countries_list.index(country)+1]})

for i, city in enumerate(cites_cases_list):
    aux_cases_list[i].update(city)
    
for case in aux_cases_list:
    case.update({
    "Tipo contrato": "",
    "Salario Bruto": 0 ,
    "Salario Neto" : 0 ,
    "Pagos en dólares": 0 ,
    "Valor de dolar tomado": 0 ,
    "Conformidad con sueldo": 0 ,
    "Conformidad con sueldo último semestre": 0 ,
    "Bono": "",
    "A qué está atado el bono": "",
    "Ajustes por inflación 2021": "",
    "% del ajuste": 0,
    "Mes de Ajuse": 0,
    "Rol": "",
    "Años de experiencia": 0,
    "Años en la empresa": 0,
    "Años en el puesto": 0,
    "Gente a cargo": 0,
    "Plataformas": 0,
    "Lenguajes de programación - tecnologías": "",
    'Frameworks - herramientas - librerías': "",
    "Bases de datos": '',
    "QA / Testing": "",
    "IDEs": "",
    "Cantidad de personas en org": 0,
    "Org con oficina en mi ciudad": "",
    "Actividad principal": "",
    "eNPS": 0,
    "Calificación de políticas de diversidad e inclusión": 0,
    "Beneficios extra": "",
    "Salir o seguir Estudios":"",
    "Nivel de estudios alcanzado": "",
    "Estado": "",
    "Carrera": "",
    "Universidad": "",
    "Cursos de especialización": "",
    "¿Contribuís a proyectos open source?": "",
    "¿Programás como hobbie?": "",
    "Salir o seguir Guardias": "",
    "¿Tenés guardias?": "",
    "Cuánto cobrás por guardia": "",
    "¿Porcentaje, bruto o neto?": "",
    "Salir o seguir Seguridad Laboral": "",
    "¿Situaciones de violencia laboral?": "",
    "¿Tenés algún tipo de discapacidad?": "",
    "Dificultad para conseguir trabajo por discapacidad": "",
    "Salir o seguir Pandemia": "",
    "¿Cómo venís llevando la pandemia?": "",
    "¿Tenés hijos/as menores de edad?": "",
    "¿Con cuántas personas estás conviviendo?": "",
    "¿Con quiénes convivís?": "",
    "¿Tenés un espacio dedicado para el trabajo?": "",
    "¿Tenés que compartir tu equipo de trabajo con alguien?": "",
    "¿Qué tipo de cuarentena hiciste / estás haciendo?": "",
    "¿Cambió tu situación laboral a raíz de la pandemia?": "",
    "Apoyo de empresa/organización en esta situación?": "",
    "¿Cómo se vio afectada tu empresa/organización?": "",
    "¿Instauraron algún beneficio nuevo?": "",
    "Edad": "",
    "Identidad Sex": "",        
        
})

aux_cases_list_keys = list(aux_cases_list[0].keys())


for i, case in enumerate(raw_cases_LATAM):
    indi = len(countries_list)
    for key  in aux_cases_list_keys[2:]:
        indi+=1
        aux_cases_list[i][key] = case [indi]


cases_list.extend(aux_cases_list)



#Elegí una lista de diccionarios para asignar el nombre de las columnas a las keys y así poder llamar la info
#todos los diccionarios tienen las mismas keys y estan almacenados en la lista "cases_list"
for case in cases_list:
    if case['Region'] == 'Chinandega':
        cases_list.remove(case)
        
#Decidí borrar este caso porque es outlier en casi todo los campos 


2. Printear porcentaje de participación por región, ordenado de mayor a menor. Ej: 

```
- Ciudad Autónoma de Buenos Aires - 59.4%
- Formosa - 0.04%
```

In [2]:
def amount_of_cases_per_item (cases_list_key):
    items_set = set()

    for case in cases_list:
        items_set.add(case[cases_list_key])

    items_list = list(items_set)
    items_list.sort()

    def item_num_cases (item):
        number_of_cases = 0
        for case in cases_list:
            if case[cases_list_key] == item:
                number_of_cases += 1
        return {item: number_of_cases}
    
    items_dic = {}

    for item in items_list:
        items_dic.update(item_num_cases (item))
    
    return items_dic

# - amount_of_cases_per_item - devuelve diccionario con la cantidad de respuestas por respuesta

def percentage_of_cases_per_item (cases_list_key):

    items_dic = amount_of_cases_per_item (cases_list_key)
    
    for key in items_dic:
        percentage = items_dic[key] * 100 / len(cases_list)
        items_dic[key] = round(percentage,2) 
    
    sorted_items_dic = dict(sorted(items_dic.items(), key=lambda item: item[1], reverse=True))
    
    return sorted_items_dic

# - percentage_of_cases_per_item - devuelve diccionario ordenado con el porcenatje de respuesta por respuesta

cities_dic = percentage_of_cases_per_item ("Region")

for key in cities_dic:
    print ("- " + key + " - " + str(cities_dic[key]) + '%')
        


- Ciudad Autónoma de Buenos Aires - 51.72%
- Provincia de Buenos Aires - 10.1%
- GBA - 9.2%
- Córdoba - 7.8%
- Santa Fe - 5.82%
- Mendoza - 2.18%
- Montevideo - 1.7%
- Entre Ríos - 0.96%
- Río Negro - 0.76%
- Tucumán - 0.76%
- Neuquén - 0.68%
- Misiones - 0.61%
- Santiago - 0.58%
- Chaco - 0.57%
- México DF - 0.47%
- Lima - 0.45%
- Salta - 0.42%
- Corrientes - 0.35%
- San Juan - 0.35%
- Bogotá - 0.28%
- Chubut - 0.28%
- Jujuy - 0.28%
- Medellín - 0.25%
- La Rioja - 0.23%
- Quito - 0.23%
- San Luis - 0.17%
- Tierra del Fuego - 0.17%
- Santiago de Cali - 0.13%
- Ciudad Guatemala - 0.12%
- La Pampa - 0.12%
- Santiago del Estero - 0.12%
- Guadalajara - 0.1%
- Asunción - 0.09%
- Guayaquil - 0.09%
- Caracas - 0.07%
- Santa Cruz - 0.07%
- Santa Cruz de la Sierra - 0.07%
- Querétaro - 0.06%
- Catamarca - 0.04%
- Cochabamba - 0.04%
- Loja - 0.04%
- Manizales - 0.04%
- Monterrey - 0.04%
- Puebla de Zaragoza - 0.04%
- San José - 0.04%
- Tunja - 0.04%
- Valencia - 0.04%
- Valparaíso - 0.04%
- Vera

3. Printear porcentaje de participación por tipo de rol, ordenado de mayor a menor. Ej:  
`- Developer - 39.11 %`

In [3]:
for case in cases_list:
    case["Rol"] = case["Rol"].upper()

for case in cases_list:
    if len (case["Rol"])>3 and case["Rol"][-1] == ' ':
        case["Rol"] = case["Rol"][0:-1]
            
    
def answer_replace_str (question, str_to_replace, replacement):
    for case in cases_list:
        if str_to_replace in case[question]:
            case[question] = case[question].replace(str_to_replace, replacement)

# - answer_replace_str - cambia un string por otro en todas las respuestas a ls pregunta que se ingrese como parametro

answer_replace_str ("Rol",'Í', 'I')
answer_replace_str ("Rol",'Ó', 'O')
answer_replace_str ("Rol",'Á', 'A')
answer_replace_str ("Rol",'É', 'E')
answer_replace_str ("Rol",'Ú', 'U')
answer_replace_str ("Rol",'Ñ', 'N')

rol_dic = percentage_of_cases_per_item ("Rol")

rol_others = 0
for key in rol_dic:
    if rol_dic[key] < 0.1:
        rol_others += rol_dic[key]


for key in rol_dic:
    if rol_dic[key] > 0.1:
        print ("- " + key + " - " + str(rol_dic[key]) + '%')
print ("- " + 'OTROS' + " - " + str(round(rol_others,2)) + '%')


- DEVELOPER - 40.7%
- SYSADMIN / DEVOPS / SRE - 11.09%
- TECHNICAL LEADER - 6.16%
- QA / TESTER - 4.3%
- MANAGER / DIRECTOR - 4.28%
- PROJECT MANAGER - 3.09%
- HELPDESK - 2.62%
- BI ANALYST / DATA ANALYST - 2.45%
- ARCHITECT - 2.23%
- CONSULTANT - 1.92%
- DATA ENGINEER - 1.76%
- BUSINESS ANALYST - 1.69%
- NETWORKING - 1.54%
- DATA SCIENTIST - 1.49%
- UX - 1.49%
- INFOSEC - 1.43%
- RECRUITER / HR - 1.24%
- PRODUCT MANAGER - 1.21%
- DBA - 0.71%
- SCRUM MASTER - 0.66%
- DESIGNER - 0.64%
- VP / C-LEVEL - 0.55%
- SALES / PRE-SALES - 0.42%
- ANALISTA FUNCIONAL - 0.19%
- FINANCE - 0.17%
- MIDDLEWARE - 0.17%
- TECHNICAL WRITER - 0.16%
- STORAGE / BACKUP - 0.12%
- OTROS - 4.33%


4. Printear porcentaje de respuesta para las preguntas: 
* ¿Contribuís a proyectos Open Source?
* ¿Programás por hobbie?

In [4]:
open_source_dic = amount_of_cases_per_item ("¿Contribuís a proyectos open source?")
hobbie_dic = amount_of_cases_per_item ("¿Programás como hobbie?")

def y_n_answer_percentages (amount_of_cases_dic):
    no_answer = 0

    for key in amount_of_cases_dic:
        if key == "":
            no_answer += amount_of_cases_dic [key]
    return {
    'Respuestas': (len(cases_list) - no_answer) * 100 / len(cases_list),
    'Porcentaje si':  amount_of_cases_dic['Sí'] * 100/ (amount_of_cases_dic['Sí'] + amount_of_cases_dic['No']),
    'Porcentaje no':  amount_of_cases_dic['No'] * 100/ (amount_of_cases_dic['Sí'] + amount_of_cases_dic['No'])}
    
# - y_n_answer_percentages - devuelve diccionario con el porcentaje de respuestas sobre el total
# y el porcentaje de si y no sobre respuestas

open_source_percent_dic = y_n_answer_percentages (open_source_dic)
hobbie_percent_dic = y_n_answer_percentages (hobbie_dic)


print(str(round(open_source_percent_dic['Respuestas'], 2)) + '% de las personas encuestadas contestaron la pregunta "¿Contribuís a proyectos Open Source?"')
print(str(round(open_source_percent_dic['Porcentaje si'], 2)) + '% de las respuestas fueron afmirmativas, ' + str(round(open_source_percent_dic['Porcentaje no'], 2)) + '% fueron negativas.')
print("""""")
print(str(round(hobbie_percent_dic['Respuestas'], 2)) + '% de las personas encuestadas contestaron la pregunta "¿Programás como hobbie?"')
print(str(round(hobbie_percent_dic['Porcentaje si'], 2)) + '% de las respuestas fueron afmirmativas, ' + str(round(hobbie_percent_dic['Porcentaje no'], 2)) + '% fueron negativas.')



50.57% de las personas encuestadas contestaron la pregunta "¿Contribuís a proyectos Open Source?"
14.97% de las respuestas fueron afmirmativas, 85.03% fueron negativas.

50.57% de las personas encuestadas contestaron la pregunta "¿Programás como hobbie?"
56.06% de las respuestas fueron afmirmativas, 43.94% fueron negativas.


5. Printear porcentaje por seniority según años de experiencia, el mapeo es:
```
Junior: de 0 hasta 2 años.
Semi-Senior: de 2 años inclusive hasta 5 años.
Senior: desde 5 años inclusive.
``` 

In [5]:
seniority_dic = amount_of_cases_per_item ("Años de experiencia")

    
def sum_by_rank (amount_dic, num1, numb2):    
    seniority_list = []

    for key in amount_dic:
        seniority_list.append((float(key), seniority_dic[key] ))

    first_rank = 0
    second_rank = 0
    third_rank = 0

    for i in seniority_list:
        if i[0] < num1:
            first_rank += i[1]
        elif i[0] >= num1 and i[0] < numb2 :
            second_rank += i[1]
        elif i[0] >= numb2 :
            third_rank += i[1]
    
    return {'first_rank': first_rank,
        'second_rank': second_rank,
         'third_rank': third_rank}
        
seniority_by_rank_dic = sum_by_rank (seniority_dic, 2, 5)
        

junior_percentage = round (seniority_by_rank_dic['first_rank'] * 100 / len(cases_list), 2) 
semi_senior_percentage = round (seniority_by_rank_dic['second_rank'] * 100 / len(cases_list), 2)
senior_percentage = round (seniority_by_rank_dic['third_rank'] * 100 / len(cases_list), 2)
        
print("El procentaje de encuestados Junior (con menos de dos años de experiencia) es: " + str(junior_percentage) + '%')       
print("El procentaje de encuestados Semi-Senior (entre 2 y 5 años de expriencia) es: " + str(semi_senior_percentage) + '%')
print("El procentaje de encuestados Senior (mas de 5 años de experiencia) es: " + str(senior_percentage) + '%')


El procentaje de encuestados Junior (con menos de dos años de experiencia) es: 15.93%
El procentaje de encuestados Semi-Senior (entre 2 y 5 años de expriencia) es: 26.63%
El procentaje de encuestados Senior (mas de 5 años de experiencia) es: 57.44%


6. Printear porcentaje de personas encuestadas por años en la compañía actual y por años en el puesto actual.

In [6]:
years_in_company_dic = amount_of_cases_per_item ("Años en la empresa")

years_in_position_dic = amount_of_cases_per_item ("Años en el puesto")



7. Printear porcentajes de nivel de educación formal y estado, es decir: % educación secundaria, terciaria, universitaria, postgrado, doctorado, postdoctorado, completo, incompleto y en curso para cada uno.

In [7]:
educational_level_set = set()

for case in cases_list:
    educational_level_set.add((case["Nivel de estudios alcanzado"], case["Estado"]))
    
educational_level_list = list(educational_level_set)
educational_level_list.sort()
educational_level_list.remove (('', ''))

educational_level_dic = {}

#number_of_cases = 0

for i in educational_level_list:
    number_of_cases = 0
    for case in cases_list:
        if case["Nivel de estudios alcanzado"] == i[0] and case["Estado"] == i[1]:
            number_of_cases += 1
    educational_level_dic.update({i:number_of_cases})

number_education_answers = 0    
for key in educational_level_dic:
    number_education_answers += educational_level_dic[key]

number_education_answers

for key in educational_level_dic:
    educational_level_dic[key] = (educational_level_dic[key] * 100) / number_education_answers 

sorted_educational_level_dic = dict(sorted(educational_level_dic.items(), key=lambda item: item[1], reverse=True))
sorted_educational_level_dic

print("Porcentaje de respuestas por nivel de educación formal sobre el total de respuestas a esta pregunta:")

for key in sorted_educational_level_dic:
    print('- ' + key[0] + ' ' + key[1] + ' - ' + str(round(sorted_educational_level_dic[key], 2)) + '%')

Porcentaje de respuestas por nivel de educación formal sobre el total de respuestas a esta pregunta:
- Universitario En curso - 25.45%
- Universitario Completado - 25.19%
- Universitario Incompleto - 17.85%
- Terciario Completado - 8.84%
- Secundario Completado - 5.79%
- Terciario En curso - 4.75%
- Posgrado Completado - 4.55%
- Terciario Incompleto - 3.71%
- Posgrado En curso - 1.64%
- Posgrado Incompleto - 0.75%
- Secundario Incompleto - 0.46%
- Doctorado Completado - 0.37%
- Doctorado En curso - 0.29%
- Secundario En curso - 0.14%
- Doctorado Incompleto - 0.06%
- Posdoctorado Completado - 0.06%
- Posdoctorado En curso - 0.03%
- Posdoctorado Incompleto - 0.03%
- Primario Incompleto - 0.03%


8. Formatear las carreras universitarias:
- Nombres capitalizados -
- Reemplazar vocales con tilde por vocales sin tilde. -
- Reemplazar `ñ` por `n` -
- *lic*, *lic.* por Licenciatura -
- *tec*, *tec.* por Tecnucatura -
* *cs, *cs.* por Ciencias -
* *ed, ed.* por Educación - 
* Transformaciones que se consideren necesarias

Printear porcentaje según carrera

In [8]:

for case in cases_list:
    case["Carrera"] = case["Carrera"].upper()

for case in cases_list:
    if len (case["Carrera"])>3 and case["Carrera"][-1] == ' ':
        case["Carrera"] = case["Carrera"][0:-1]


answer_replace_str ("Carrera",'Í', 'I')
answer_replace_str ("Carrera",'Ó', 'O')
answer_replace_str ("Carrera",'Á', 'A')
answer_replace_str ("Carrera",'É', 'E')
answer_replace_str ("Carrera",'Ú', 'U')
answer_replace_str ("Carrera",'Ñ', 'N')
answer_replace_str ("Carrera",'LIC.', 'LICENCIATURA')
answer_replace_str ("Carrera",'LIC ', 'LICENCIATURA ')
answer_replace_str ("Carrera",'LIC,', 'LICENCIATURA')
answer_replace_str ("Carrera",'LICENCIADO', 'LICENCIATURA')
answer_replace_str ("Carrera",'TEC,', 'TECNICATURA')
answer_replace_str ("Carrera",'TEC.', 'TECNICATURA')
answer_replace_str ("Carrera",'TEC ', 'TECNICATURA ')
answer_replace_str ("Carrera",'TECNICO', 'TECNICATURA')
answer_replace_str ("Carrera",'CS.', 'CIENCIAS')
answer_replace_str ("Carrera",'CS,', 'CIENCIAS')
answer_replace_str ("Carrera",'CS ', 'CIENCIAS ')
answer_replace_str ("Carrera",'ED.', 'EDUCACION')
answer_replace_str ("Carrera",'ED,', 'EDUCACION')
answer_replace_str ("Carrera",'ED ', 'EDUCACION ')
answer_replace_str ("Carrera",'ING.', 'INGENIERIA')
answer_replace_str ("Carrera",'ING ', 'INGENIERIA ')
answer_replace_str ("Carrera",'ING,', 'INGENIERIA')
answer_replace_str ("Carrera",'-', '')
answer_replace_str ("Carrera",'.', '')

career_dic = amount_of_cases_per_item ("Carrera")

n_career_answers = len(cases_list) - career_dic['']

del(career_dic[''])

sorted_career_dic = dict(sorted(career_dic.items(), key=lambda item: item[1], reverse=True))

for item in sorted_career_dic:
    sorted_career_dic[item] = (sorted_career_dic[item] * 100) / n_career_answers

career_others = 0
for key in sorted_career_dic:
    if sorted_career_dic[key] < 0.1:
        career_others += sorted_career_dic[key]

print("Porcentaje porcentaje según carrera sobre el total de respuestas a esta pregunta:")


for key in sorted_career_dic:
    if sorted_career_dic[key] > 0.1:
        print ("- " + key + " - " + str(round(sorted_career_dic[key], 2)) + '%')
print ("- " + 'OTROS' + " - " + str(round(career_others,2)) + '%')



Porcentaje porcentaje según carrera sobre el total de respuestas a esta pregunta:
- INGENIERIA EN SISTEMAS DE INFORMACION - 19.01%
- INGENIERIA EN INFORMATICA - 11.35%
- ANALISTA DE SISTEMAS - 8.69%
- LICENCIATURA EN CIENCIAS DE LA COMPUTACION - 5.66%
- LICENCIATURA EN SISTEMAS DE INFORMACION - 5.66%
- LICENCIATURA EN INFORMATICA - 4.3%
- TECNICATURA EN PROGRAMACION - 3.39%
- INGENIERIA ELECTRONICA - 3.0%
- INGENIERIA EN COMPUTACION - 2.82%
- DISENO GRAFICO - 2.48%
- TECNICATURA SUPERIOR EN PROGRAMACION - 2.36%
- LICENCIATURA EN ADMINISTRACION - 1.73%
- TECNICATURA EN INFORMATICA - 1.57%
- INGENIERIA INDUSTRIAL - 1.18%
- LICENCIATURA EN ANALISIS DE SISTEMAS - 1.18%
- ANALISTA PROGRAMADOR - 1.12%
- INGENIERIA EN SOFTWARE - 1.06%
- ANALISTA PROGRAMADOR UNIVERSITARIO - 1.0%
- INGENIERIA EN TELECOMUNICACIONES - 0.97%
- ANALISTA DE COMPUTACION - 0.88%
- TECNICATURA SUPERIOR EN SISTEMAS INFORMATICOS - 0.88%
- DISENO MULTIMEDIAL - 0.85%
- TECNICATURA EN SISTEMAS - 0.85%
- TECNICATURA SUPERIOR

9. Printear porcentaje de respuesta a las preguntas:
- ¿Realizaste cursos de especialización?
- Si realizaste, ¿Quién pagó por los cursos de especialización?

In [9]:
courses_dic = amount_of_cases_per_item ("Cursos de especialización")

n_courses_answers = len(cases_list) - courses_dic['']
no_percentage = (courses_dic['No'] * 100)/n_courses_answers

prticular_amount = 0
for key in courses_dic:
    if 'particular' in key:
        prticular_amount += courses_dic[key]
        
employer_amount = 0
for key in courses_dic:
    if 'empleador' in key:
        employer_amount += courses_dic[key]
        
prticular_percentage = (prticular_amount * 100)/n_courses_answers
employer_percentage = (employer_amount * 100)/n_courses_answers


print(str(round (no_percentage,2)) + '% de las pesonas que respondieron no hicieron ningún curso de especialización' )

print(str(round (prticular_percentage,2)) + '% de las pesonas que respondieron hicieron cursos de especialización de forma particular' )

print(str(round (employer_percentage,2)) + '% de las pesonas que respondieron hicieron cursos de especialización pagados por un empleador' )


20.88% de las pesonas que respondieron no hicieron ningún curso de especialización
70.57% de las pesonas que respondieron hicieron cursos de especialización de forma particular
29.02% de las pesonas que respondieron hicieron cursos de especialización pagados por un empleador


10. Printear porcentaje por identidad de género y personas con discapacidad

In [10]:
gender_dic = percentage_of_cases_per_item ("Identidad Sex")
gender_unexpected = 100 - (gender_dic['Varón Cis'] + gender_dic['Mujer Cis'] + gender_dic['No binarie'] + gender_dic['Varón Trans'] + gender_dic['Mujer Trans'])

print(str(gender_dic['Varón Cis']) + '% de los encuestados se identifican como varones cis')
print(str(gender_dic['Mujer Cis']) + '% de los encuestados se identifican como mujeres cis')
print(str(gender_dic['No binarie']) + '% de los encuestados se identifican como no binaries')
print(str(gender_dic['Varón Trans']) + '% se identifican como varones trans y ' + str(gender_dic['Mujer Trans']) + '% como mujeres trans')
print(str (round(gender_unexpected, 2)) + "% de las respuestas no entran en ninguna categoría de las anteriores")

#####################################
# Fin de identidad sex
#####################################


for case in cases_list:
    case["¿Tenés algún tipo de discapacidad?"] = case["¿Tenés algún tipo de discapacidad?"].upper()


answer_replace_str ("¿Tenés algún tipo de discapacidad?",',', '')
answer_replace_str ("¿Tenés algún tipo de discapacidad?",'NINGUNA', '')
answer_replace_str ("¿Tenés algún tipo de discapacidad?",'NINGUNO ', '')
answer_replace_str ("¿Tenés algún tipo de discapacidad?",'NO APLICA', '')

for case in cases_list:
    if len(case["¿Tenés algún tipo de discapacidad?"]) < 3:
        case["¿Tenés algún tipo de discapacidad?"] = ''


disabilities_dic = percentage_of_cases_per_item ("¿Tenés algún tipo de discapacidad?")    
    
print("""
""")

print ("- " + "No tienen ninguna discapacidad" + " - " + str(disabilities_dic['']) + '%')    

for key in disabilities_dic:
    if key != '':
        print ("- " + key + " - " + str(round(disabilities_dic[key], 2)) + '%')


75.25% de los encuestados se identifican como varones cis
16.16% de los encuestados se identifican como mujeres cis
0.73% de los encuestados se identifican como no binaries
0.2% se identifican como varones trans y 0.1% como mujeres trans
7.56% de las respuestas no entran en ninguna categoría de las anteriores


- No tienen ninguna discapacidad - 96.62%
- VISUAL - 2.26%
- AUDITIVA - 0.19%
- MENTAL - 0.19%
- MOTRIZ - 0.09%
- AUDITIVA VISUAL - 0.04%
- VISCERAL - 0.04%
- ADHD - 0.03%
- EPILEPSIA - 0.03%
- OBESIDAD - 0.03%
- TDAH - 0.03%
- USO ANTEOJOS - 0.03%
- APATÍA SELECTIVA - 0.01%
- ASPERGER - 0.01%
- AUDICION DISMINUIDA DE UN OIDO SUFICIENTE COMO PARA NO PODER TRABAJAR EN CALLCENTER PERO MINIMA COMO PARA QUE NO LES IMPORTE A MI EMPLEADOR ACTUAL - 0.01%
- AUDITIVA MENTAL - 0.01%
- AUTISMO - 0.01%
- AUTISMO (?) - 0.01%
- AUTUSMO - 0.01%
- DALTONISMO (DEUTEROTOPIA) - 0.01%
- DE NO ENTENDER COMO EN LA RESPUESTA ANTERIOR NO PUSIERON UNA RESPUESTA LIBRE. SOY FURRO POR ELECCION Y CONVICCION

11. Salarios: calcular la mediana salarial para cada una de las siguientes categorías:
- Salario según región
- Salario por rol
- Salario por experiencia
- Salario por nivel de formación
- Salario por carrera
- Salario por tecnología
- Salario por lenguaje de programación
- Salario por género

In [11]:

for case in cases_list:
    if case['Salario Neto'] != '':
        case['Salario Neto'] = float(case['Salario Neto'])

for case in cases_list:
    if case['Valor de dolar tomado'] != '':
        try:
            case['Valor de dolar tomado'] = float(case['Valor de dolar tomado'])
        except ValueError:
            case['Valor de dolar tomado'] = ''

for case in cases_list:
    case.update({'Salario Neto en US$': ''})

for case in cases_list:
    if case['Valor de dolar tomado'] == 0:
        case['Valor de dolar tomado'] = ''

for case in cases_list:
    if case['Valor de dolar tomado'] != '' and case['Salario Neto'] != '':
        case['Salario Neto en US$'] = case['Salario Neto'] / case['Valor de dolar tomado'] 

        
dollar_countries_list = ['Puerto Rico', 'Panamá', 'Islas Vírgenes', 'El Salvador', 'Ecuador', 'Bahamas']       
        
for case in cases_list:
    if case['Salario Neto'] != '' and case['Pais'] in dollar_countries_list:
        case['Salario Neto en US$'] = case['Salario Neto'] 

        
currencies_dic = {'Antigua y Barbuda': 2.70, #XCD Dolar del caribe oriental
                'Argentina': 104.30, #ARS
                'Aruba': 1.79, #AWG
                'Barbados': 2, #BBD
                'Belice': 2.0094, #BZD
                'Bolivia': 6.90, #BOB
                'Brasil': 5.55, #BRL
                'Cuba': 24, #CUP
                'Colombia': 3960, #COP
                'Costa Rica': 628.07, #CRC
                'Chile': 813.5, #CLP
                'Granada': 2.7, #XCD
                'Guadalupe': 0.88, #EUR
                'Guatemala': 7.73, #GTQ
                'Guyana': 209.02, #GYD
                'Guyana Francesa': 0.88, #EUR
                'Haití': 104, #HTG
                'Honduras': 24.12, #HNL 
                'Islas Caimán': 0.83, #KYD
                'Jamaica': 155.68, #JMD
                'Martinica': 0.88, #EUR
                'México': 20.61, #MXN
                'Nicaragua': 35.35, #NIO
                'Paraguay': 6975.32, #PYG
                'Perú': 3.83, #PEN
                'República Dominicana': 57.80, #DOP
                'San Cristóbal y Nieves': 2.70, #XCD
                'San Vicente y las Granadinas': 2.70, #XCD 
                'Santa Lucía': 2.70, #XCD 
                'Suriname': 21, #SRD
                'Trinidad y Tobago': 6.78, #TTD
                'Uruguay': 44.67, #UYU
                'Venezuela': 460550, } #VEF


def local_currency_to_us (case, to_convert):
    usdollar = 0
    for key in currencies_dic:
        if case["Pais"] == key and case[to_convert] != '':
            usdollar = round(case[to_convert] / currencies_dic[key], 2)
    return(usdollar)

# - local_currency_to_us - utiliza el diccionario de monedas para convertir a dolares todas las monedas de latam

for case in cases_list:
    if case['Pagos en dólares'] == '':
        case['Salario Neto en US$'] = local_currency_to_us (case, 'Salario Neto')
        
for case in cases_list:
    if case['Pagos en dólares'] == '':
        case['Salario Neto en US$'] = local_currency_to_us (case, 'Salario Neto')


#########################################
# FIN DE CONVERSION DE SUELDOS A DOLARES       
#########################################       


In [12]:
for case in cases_list:
    if case['Salario Neto en US$'] != '' and case['Salario Neto en US$'] < 10:
        case['Salario Neto en US$'] = ''
#Para este analisis voy asumir que los sueldos menores a 10US$ son un error y los voy a descartar 

def salaries_median_by_item (question):
    items_set = set()

    for case in cases_list:
        items_set.add(case[question])

    items_list = list(items_set)
    items_list.sort()

    salaries_lists_dic = {}

    for i in items_list:
        salaries_lists_dic.update({i:[]})

    for key in salaries_lists_dic:
        for case in cases_list:
            if key == case[question] and case['Salario Neto en US$'] != '':
                salaries_lists_dic[key].append(case['Salario Neto en US$'])

    for key in salaries_lists_dic:
        salaries_lists_dic[key].sort()

    salaries_median_dic = {}

    for key in salaries_lists_dic:
        if len(salaries_lists_dic[key]) % 2 ==0 and salaries_lists_dic[key] != []:
            num1 = salaries_lists_dic[key][int((len(salaries_lists_dic[key]) / 2) - 1)]
            num2 = salaries_lists_dic[key][int(len(salaries_lists_dic[key]) / 2)]
            mediana = (num1 + num2) / 2
            salaries_median_dic.update({key: (round(mediana,2), len(salaries_lists_dic[key]))})


    for key in salaries_lists_dic:
        if len(salaries_lists_dic[key]) % 2 !=0 and salaries_lists_dic[key] != []:
            mediana = salaries_lists_dic[key][int(len(salaries_lists_dic[key]) / 2)]
            salaries_median_dic.update({key: (round(mediana,2), len(salaries_lists_dic[key]))})

    return (salaries_median_dic)

# - salaries_median_by_item - devuelve un diccionario con la mediana de salarios por item en una pregunta 
# y la cantidad de casos encontrados



In [13]:
region_salary = salaries_median_by_item ('Region')

sorted_region_salary = dict(sorted(region_salary.items(), key=lambda item: item[1], reverse=True))

print("""Salario según región:
""")



for key in sorted_region_salary:
    if key != '' and sorted_region_salary[key][1] > 10:
        print ("- " + key + " - " + 'mediana salarial US$' + str(sorted_region_salary[key][0]))

#Voy a tener en cuenta las respuestas con mas de diez casos 
#Voy a usar el mismo criterio en las siguientes respuestas

Salario según región:

- Santiago - mediana salarial US$2243.39
- México DF - mediana salarial US$1843.77
- Montevideo - mediana salarial US$1756.62
- Medellín - mediana salarial US$1464.65
- Quito - mediana salarial US$1325.0
- Ciudad Autónoma de Buenos Aires - mediana salarial US$1246.4
- Tucumán - mediana salarial US$1199.24
- GBA - mediana salarial US$1150.53
- Santa Fe - mediana salarial US$1150.53
- Chubut - mediana salarial US$1142.77
- Bogotá - mediana salarial US$1136.36
- Río Negro - mediana salarial US$1124.7
- Provincia de Buenos Aires - mediana salarial US$1122.34
- Córdoba - mediana salarial US$1113.13
- Jujuy - mediana salarial US$1015.34
- Neuquén - mediana salarial US$1006.71
- Mendoza - mediana salarial US$1003.36
- La Rioja - mediana salarial US$909.99
- Entre Ríos - mediana salarial US$865.23
- San Juan - mediana salarial US$855.1
- Lima - mediana salarial US$835.51
- Salta - mediana salarial US$771.82
- Misiones - mediana salarial US$759.85
- Chaco - mediana salari

In [14]:
rol_salary = salaries_median_by_item ('Rol')

sortedrol_salary = dict(sorted(rol_salary.items(), key=lambda item: item[1], reverse=True))

print("""Salario por rol
""")

for key in sortedrol_salary:
    if sortedrol_salary[key][1] > 10:
        print ("- " + key + " - " + 'mediana salarial US$' + str(sortedrol_salary[key][0]))
        


Salario por rol

- VP / C-LEVEL - mediana salarial US$2703.74
- MANAGER / DIRECTOR - mediana salarial US$1927.13
- TECHNICAL LEADER - mediana salarial US$1649.09
- ARCHITECT - mediana salarial US$1629.91
- PRODUCT MANAGER - mediana salarial US$1581.98
- SCRUM MASTER - mediana salarial US$1517.74
- PROJECT MANAGER - mediana salarial US$1489.93
- DBA - mediana salarial US$1342.28
- DATA SCIENTIST - mediana salarial US$1294.34
- DATA ENGINEER - mediana salarial US$1271.11
- SALES / PRE-SALES - mediana salarial US$1251.2
- INFOSEC - mediana salarial US$1246.4
- SYSADMIN / DEVOPS / SRE - mediana salarial US$1204.03
- UX - mediana salarial US$1188.88
- DEVELOPER - mediana salarial US$1150.53
- BUSINESS ANALYST - mediana salarial US$1150.53
- FINANCE - mediana salarial US$1073.83
- BI ANALYST / DATA ANALYST - mediana salarial US$1006.71
- CONSULTANT - mediana salarial US$1003.36
- QA / TESTER - mediana salarial US$968.36
- NETWORKING - mediana salarial US$968.36
- RECRUITER / HR - mediana sal

In [15]:
study_salary = salaries_median_by_item ('Nivel de estudios alcanzado')

sorted_study_salary = dict(sorted(study_salary.items(), key=lambda item: item[1], reverse=True))


print("""Salario por nivel de formación 
""")


for key in sorted_study_salary:
    if key != '' and sorted_study_salary[key][1] > 10:
        print ("- " + key + " - " + 'mediana salarial US$' + str(sorted_study_salary[key][0]))


Salario por nivel de formación 

- Doctorado - mediana salarial US$1489.33
- Posgrado - mediana salarial US$1450.98
- Universitario - mediana salarial US$1198.47
- Secundario - mediana salarial US$1083.41
- Terciario - mediana salarial US$1054.65


In [16]:
career_salary = salaries_median_by_item ('Carrera') 

sorted_career_salary = dict(sorted(career_salary.items(), key=lambda item: item[1], reverse=True))

print("""Salario por carrera
""")


for key in sorted_career_salary:
    if key != '' and sorted_career_salary[key][1] > 10:
        print ("- " + key + " - " + 'mediana salarial US$' + str(sorted_career_salary[key][0]))



Salario por carrera

- LICENCIATURA EN ANALISIS DE SISTEMAS - mediana salarial US$1529.39
- INGENIERIA EN TELECOMUNICACIONES - mediana salarial US$1395.01
- INGENIERIA EN SOFTWARE - mediana salarial US$1386.55
- LICENCIATURA EN CIENCIAS DE LA COMPUTACION - mediana salarial US$1347.07
- INGENIERIA EN INFORMATICA - mediana salarial US$1342.28
- ANALISTA PROGRAMADOR UNIVERSITARIO - mediana salarial US$1271.81
- ANALISTA UNIVERSITARIO DE SISTEMAS - mediana salarial US$1259.93
- INGENIERIA EN SISTEMAS DE INFORMACION - mediana salarial US$1246.4
- LICENCIATURA EN SISTEMAS DE INFORMACION - mediana salarial US$1246.4
- INGENIERIA EN COMPUTACION - mediana salarial US$1231.9
- LICENCIATURA EN INFORMATICA - mediana salarial US$1198.47
- INGENIERIA INDUSTRIAL - mediana salarial US$1198.47
- DISENO GRAFICO - mediana salarial US$1191.28
- TECNICATURA SUPERIOR EN SISTEMAS INFORMATICOS - mediana salarial US$1179.29
- LICENCIATURA EN ADMINISTRACION - mediana salarial US$1164.91
- ANALISTA DE SISTEMAS -

In [17]:
gender_salary = salaries_median_by_item ('Identidad Sex')


sorted_gender_salary = dict(sorted(gender_salary.items(), key=lambda item: item[1], reverse=True))

gender_identities = ['Varón Trans', 'Varón Cis', 'Mujer Trans', 'Mujer Cis', 'No binarie']


print("""Salario por género
""")


for key in sorted_gender_salary:
    if key in gender_identities and sorted_gender_salary[key][1] > 10:
        print ("- " + key + " - " + 'mediana salarial US$' + str(sorted_gender_salary[key][0]))


Salario por género

- Varón Trans - mediana salarial US$1524.45
- Varón Cis - mediana salarial US$1231.41
- Mujer Cis - mediana salarial US$997.92
- No binarie - mediana salarial US$970.4


In [18]:
seniority_salary_lst = []

for case in cases_list:
    if case['Salario Neto en US$'] != "":
        seniority_salary_lst.append((float(case["Años de experiencia"]),case['Salario Neto en US$']))

salary_seniority_dic = {"junior_0_2": [],
                   'semi_senior_2_5': [],
                   'senior_5': []}

for case in seniority_salary_lst:
    if case[0] < 2:
        salary_seniority_dic['junior_0_2'].append(case[1])
    elif case[0] >= 2 and case[0] < 5:
        salary_seniority_dic['semi_senior_2_5'].append(case[1])
    elif case[0] >= 5 :
        salary_seniority_dic['senior_5'].append(case[1])
    
salary_seniority_dic['junior_0_2'].sort()
salary_seniority_dic['semi_senior_2_5'].sort()    
salary_seniority_dic['senior_5'].sort()

def salary_median_dic (salaries_lists_dic):
    
    salary_median_dic = {}

    for key in salaries_lists_dic:
        if len(salaries_lists_dic[key]) % 2 ==0 and salaries_lists_dic[key] != []:
            num1 = salaries_lists_dic[key][int((len(salaries_lists_dic[key]) / 2) - 1)]
            num2 = salaries_lists_dic[key][int(len(salaries_lists_dic[key]) / 2)]
            mediana = (num1 + num2) / 2
            salary_median_dic.update({key: (round(mediana,2), len(salaries_lists_dic[key]))})

    for key in salaries_lists_dic:
        if len(salaries_lists_dic[key]) % 2 !=0:
            mediana = salaries_lists_dic[key][int(len(salaries_lists_dic[key]) / 2)]
            salary_median_dic.update({key: (round(mediana,2), len(salaries_lists_dic[key]))})
    return salary_median_dic

salary_seniority_median_dic = salary_median_dic (salary_seniority_dic)

print("""Salario por experiencia
""")

print('- Junior (hasta 2 años de experiencia) - mediana salarial US$' + str(salary_seniority_median_dic['junior_0_2'][1]))
print('- Semi Senior (entre 2 y 5 años de experiencia) - mediana salarial US$' + str(salary_seniority_median_dic['semi_senior_2_5'][1]))
print('- Senior (mas de 5 experiencia) - mediana salarial US$' + str(salary_seniority_median_dic['senior_5'][1]))




Salario por experiencia

- Junior (hasta 2 años de experiencia) - mediana salarial US$933
- Semi Senior (entre 2 y 5 años de experiencia) - mediana salarial US$1561
- Senior (mas de 5 experiencia) - mediana salarial US$3256


In [19]:

tecnology_cases_list = []
for case in cases_list:
    tecnology_cases_list.append(case["Lenguajes de programación - tecnologías"].split(", "))

    
answer_replace_str ("Lenguajes de programación - tecnologías",'-', '')
answer_replace_str ("Lenguajes de programación - tecnologías",'.', '')


    
tecnology_set = set()

for lst in tecnology_cases_list:
    for i in lst:
        tecnology_set.add(i)

tecnologies_list = tecnology_set

salaries_by_tec_dic = {}

for tec in tecnologies_list:
    salaries_by_tec_dic.update({tec:[]})

for case in cases_list:
    for key in salaries_by_tec_dic:
        if key in case["Lenguajes de programación - tecnologías"] and case['Salario Neto en US$'] != "" :
            salaries_by_tec_dic[key].append(case['Salario Neto en US$'])
            
salaries_by_tec_dic = salary_median_dic (salaries_by_tec_dic)
            
sorted_salaries_by_tec = dict(sorted(salaries_by_tec_dic.items(), key=lambda item: item[1], reverse=True))


print("""Salario por Lenguajes de programación - tecnologías
""")

for key in sorted_salaries_by_tec:
    if sorted_salaries_by_tec[key][1] > 10 and key !="":
        print ("- " + key + " - " + 'mediana salarial US$' + str(sorted_salaries_by_tec[key][0]) + " - " + str(sorted_salaries_by_tec[key][1]))


Salario por Lenguajes de programación - tecnologías

- ABAP - mediana salarial US$3547.46 - 69
- Ninguno de los anteriores - mediana salarial US$3259.83 - 897
- Go - mediana salarial US$3076.92 - 357
- Dart - mediana salarial US$2376.24 - 41
- Bash/Shell - mediana salarial US$2329.82 - 912
- Apex - mediana salarial US$2205.18 - 11
- Ruby - mediana salarial US$2183.39 - 134
- Assembler - mediana salarial US$2109.3 - 21
- Powershell - mediana salarial US$1773.73 - 20
- no - mediana salarial US$1683.05 - 948
- Swift - mediana salarial US$1649.09 - 71
- C++ - mediana salarial US$1625.0 - 164
- Scala - mediana salarial US$1524.45 - 70
- Delphi - mediana salarial US$1438.16 - 22
- a - mediana salarial US$1419.13 - 4394
- ActionScript - mediana salarial US$1408.43 - 14
- Perl - mediana salarial US$1380.63 - 62
- Java - mediana salarial US$1284.76 - 2879
- Kotlin - mediana salarial US$1246.4 - 203
- M - mediana salarial US$1208.05 - 1663
- V - mediana salarial US$1139.72 - 180
- HTML - mediana

12. Printear porcentaje y salario según tipos de contrato .

In [20]:
contract_salaries_dic = salaries_median_by_item ('Tipo contrato')

total_contract_answers = 0

for key in contract_salaries_dic:
    total_contract_answers += contract_salaries_dic[key][1]

for key in contract_salaries_dic:
    print("- " + key + " - " + str(round((contract_salaries_dic[key][1] * 100) /total_contract_answers,2)) + "%")
    print("- " + key + '- mediana salarial US$' + str(contract_salaries_dic[key][0]))
    


- Freelance - 2.28%
- Freelance- mediana salarial US$1035.47
- Full-Time - 85.97%
- Full-Time- mediana salarial US$1198.47
- Part-Time - 4.75%
- Part-Time- mediana salarial US$549.65
- Participación societaria en una cooperativa - 0.26%
- Participación societaria en una cooperativa- mediana salarial US$1107.14
- Remoto (empresa de otro país) - 3.74%
- Remoto (empresa de otro país)- mediana salarial US$2812.5
- Tercerizado (trabajo a través de consultora o agencia) - 3.01%
- Tercerizado (trabajo a través de consultora o agencia)- mediana salarial US$1035.47


# Entrega final

**(Se requiere el notebook "Análisis de sueldos" resuelto)**

En base a los resultados obtenidos en *"Análisis de sueldos"* confeccionar conclusiones respecto a:

- Rol vs sueldos
- Nivel de educación alcanzada vs sueldos
- Género vs sueldos
- Tecnologías populares

**Conclusiones con respecto a la mediana de sueldos por roles:**
- Los roles que tiene un componente de liderazgo son los mejores pagos. Tanto los puestos con personal a cargo, como los que implican toma de desiciones, aparecen con medianas salariales superiores.
- Los siguientes mejor pagos son los roles relacionados con el análisis de datos
- en tercer lugar podríamos hubicar a los roles relacionados con la parte comercial
- en cuarto lugar aparecen roles encargados se la seguridad de la información y de la administración de sistemas 
- La ultima franja entre los mejores pagos son los especialistas en UX, desarrolladores  y roles relacionados con finanzas y analisis de negocio 

**Conclusiones con respecto a la mediana de sueldos por nivel educativo:**
- El nivel educativo influye en la mediana salarial de los casos en esta encuesta. 
- Cuanto mayor es el nivel de estudios alcanzados o en curso mayor es la mediana salarial

**Género vs sueldos**
- Los resultados de la encuesta con respecto al género/identidad sexual son dificiles de analizar, ya que el 75.25% de los encuestados se identifican como varones cis. Pordríamos decir que hay una sobre representación de Varones cis y muy pocos datos del resto de las identidades sexuales.
-  El 16.16% de las encuestadas que se identifican como mujeres cis tienen una mediana salarial notoriamente mas baja que los que se identifican como Varones Cis
- Los resultados obtenidos sobre personas trans y no Binaries se obtuvieron sobre muy pocos casos y no son concluyentes 


**Tecnologías populares**

Las diez tecnologías mas populares son:
- A
- Java
- Javascript
- SQL
- C
- M
- HTML
- Python
- CSS
- Bash/Shell





Generar gráficos de barra para mostrar los resultados mencionados anteriormente usando strings, ej:

```
developer | -----------------------------------------------
sysadmin  | ---------------------------
QA        | ------------------
          | -----------------5%--------10%----------------40%
```

In [21]:
print ("""Grafico Sueldo vs Rol
""")



for key in sortedrol_salary:
    if sortedrol_salary[key][1] > 10:
        _num = int (sortedrol_salary[key][0]/100)
        space_sum = 25 - len(key)
        print ("- " + key + (" " * space_sum) + " |" + ("-"* _num) + str( round(sortedrol_salary[key][0]/1000,2)) + "K US$") 



Grafico Sueldo vs Rol

- VP / C-LEVEL              |---------------------------2.7K US$
- MANAGER / DIRECTOR        |-------------------1.93K US$
- TECHNICAL LEADER          |----------------1.65K US$
- ARCHITECT                 |----------------1.63K US$
- PRODUCT MANAGER           |---------------1.58K US$
- SCRUM MASTER              |---------------1.52K US$
- PROJECT MANAGER           |--------------1.49K US$
- DBA                       |-------------1.34K US$
- DATA SCIENTIST            |------------1.29K US$
- DATA ENGINEER             |------------1.27K US$
- SALES / PRE-SALES         |------------1.25K US$
- INFOSEC                   |------------1.25K US$
- SYSADMIN / DEVOPS / SRE   |------------1.2K US$
- UX                        |-----------1.19K US$
- DEVELOPER                 |-----------1.15K US$
- BUSINESS ANALYST          |-----------1.15K US$
- FINANCE                   |----------1.07K US$
- BI ANALYST / DATA ANALYST |----------1.01K US$
- CONSULTANT                |

In [22]:
print("""Gráfico salario por nivel de formación 
""")


for key in sorted_study_salary:
    if key != '' and sorted_study_salary[key][1] > 10:
        _num = int (sorted_study_salary[key][0]/100)
        space_sum = 13 - len(key)
        print ("- " + key + (" " * space_sum) + " |" + ("-"* _num) + str( round(sorted_study_salary[key][0]/1000,2)) + "K US$") 



Gráfico salario por nivel de formación 

- Doctorado     |--------------1.49K US$
- Posgrado      |--------------1.45K US$
- Universitario |-----------1.2K US$
- Secundario    |----------1.08K US$
- Terciario     |----------1.05K US$


In [23]:
print ("""Gráfico Sueldo vs Género
""")


sorted_gender_salary

analyzed_gender_identities = ['Varón Cis', 'Mujer Cis']



for key in sorted_gender_salary:
    if key in analyzed_gender_identities:
        _num = int (sorted_gender_salary[key][0]/100)
        print ("- " + key + " |" + ("-"* _num) + str( round(sorted_gender_salary[key][0]/1000,2)) + "K US$") 




Gráfico Sueldo vs Género

- Varón Cis |------------1.23K US$
- Mujer Cis |---------1.0K US$


In [24]:
print("""Gráfico de porcentaje de casos por Lenguajes de programación - tecnologías
""")

sorted_salaries_by_tec

prercentage_cases_tec = {}

for key in sorted_salaries_by_tec:
    prercentage_cases_tec.update({key: round((sorted_salaries_by_tec[key][1] * 100)/len(cases_list),2)})

sorted_prercentage_cases_tec = dict(sorted(prercentage_cases_tec.items(), key=lambda item: item[1], reverse=True))


for key in sorted_prercentage_cases_tec:
    if sorted_prercentage_cases_tec[key] > 10 and key !="" and key !="no":
        _num = int (sorted_prercentage_cases_tec[key])
        space_sum = 25 - len(key)
        print ("- " + key + (" " * space_sum) + " |" + ("-"* _num) + str( sorted_prercentage_cases_tec[key]) + "%") 

 #       print ("- " + key + " - " + str(sorted_prercentage_cases_tec[key]) )


Gráfico de porcentaje de casos por Lenguajes de programación - tecnologías

- a                         |---------------------------------------------------------------63.98%
- Java                      |-----------------------------------------41.92%
- Javascript                |---------------------------------33.42%
- SQL                       |-------------------------------31.06%
- C                         |-------------------------25.19%
- M                         |------------------------24.21%
- HTML                      |-----------------------23.85%
- Python                    |---------------------21.05%
- CSS                       |----------------16.02%
- Bash/Shell                |-------------13.28%
- Ninguno de los anteriores |-------------13.06%
- TypeScript                |-----------11.74%
- PHP                       |-----------11.04%
